File: api-echo/internal/app/features/health/handlers/check_health_handler.go
package handlers

import (
	"net/http"

	"github.com/labstack/echo/v4"
	"github.com/rs/zerolog"

	"filmogophery/internal/app/routers"
)

type (
	checkHealthHandler struct {
	}
)

func NewCheckHealthHandler() routers.IRoute {
	return &checkHealthHandler{}
}

func (h *checkHealthHandler) RequireAuth() bool {
	return false
}

func (h *checkHealthHandler) Register(g *echo.Group) {
	g.GET("/health", h.handle)
}

func (h *checkHealthHandler) handle(c echo.Context) error {
	log := zerolog.Ctx(c.Request().Context())
	log.Info().Msg("accessed GET health")

	return c.JSON(http.StatusOK, "system all green")
}

File: api-hono/src/app/health/handlers/check-health.handler.ts
import { AppType } from "@/core/app";
import { StatusCodes } from "http-status-codes";

export default function (app: AppType) {
  app.get(
    `/health`,

    async (c) => {
      const result = {
        message: "system all green",
      };
      return c.json(result, StatusCodes.OK);
    },
  );
}
